### PR TITLE
Header Section Refactor - New UI

### PR DESCRIPTION
**Pull request by priscilawebdev**
Type: Refactor

The following has been addressed in the PR:

[#955](https://github.com/verdaccio/verdaccio/issues/955)
[#986](https://github.com/verdaccio/verdaccio/issues/986)

Description: "Create new App Bar "

* Done:
 - [x] Created css variables in JS.
 - [x] Added JS helpers.
 - [x] Replaced old sidebar by the "APP Bar" component from Material UI.
 - [x] Replaced logo by the new version.
 - [x] Added Avatar icon.
 - [x] Added Info Icon.
 - [x] Added Information Dialog, where the user can 'copy to the clipboard' the verdaccio commands.
 - [x] Added Verdaccio website documentation link.
 - [x] Added Drop Down Menu on the right side with a logout option

* To do:
 - [ ] Refactor Tests using Material UI helper (createShallow).
 - [ ] Add search bar in the sidebar. It will not be displayed in mobile version.
 - [ ] Add option to add avatar image (upload).
 - [ ] Add option to change the logo (upload).
 - [ ] Make it really responsive.
 - [ ] Move all css variables to JS.
 -  ~~Add i18n~~. (Future ticket)

* Test:
- [x] Check login modal UI tests

### CODE REVIEW

**Comment by ayusharma** [comment]
Please check the commit history.

**Comment by priscilawebdev** [comment]
@ayusharma fixed.

File: src/webui/components/Header/index.js [hunk 1]
```diff
@@ -1,69 +1,179 @@
...
+  constructor() {
+    super();
+    this.handleMenu = this.handleMenu.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpenInfoDialog = this.handleOpenInfoDialog.bind(this);
+    this.handleCloseInfoDialog = this.handleCloseInfoDialog.bind(this);
+    this.handleToggleLogin = this.handleToggleLogin.bind(this);
+    this.renderInfoDialog = this.renderInfoDialog.bind(this);
+    this.state = {
+      anchorEl: null,
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: I suggest full name `menuAnchorElement`.

File: src/webui/components/Header/index.js [hunk 2]
```diff
@@ -1,69 +1,179 @@
...
+    const {toggleLoginModal} = this.props;
+    this.setState(
+      {
+        anchorEl: null,
+      },
+      () => toggleLoginModal()
+    );
+  }
+
+  renderLeftSide() {
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: Please think good name. `renderVerdaccioLogo`

File: src/webui/components/Header/index.js [hunk 3]
```diff
@@ -1,69 +1,179 @@
...
+        <Tooltip title="Copy to Clipboard">
+          <CopyIcon aria-label="Copy to Clipboard" onClick={copyToClipBoard(text)}>
+            <FileCopy />
+          </CopyIcon>
+        </Tooltip>
+      </ClipBoardCopy>
+    );
+  }
+
+  renderInfoDialog() {
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: may be `rednerRegistryInformationDialog` ?

File: src/webui/components/Header/index.js [hunk 4]
```diff
@@ -1,69 +1,179 @@
...
+  }
+
+  renderMenu() {
+    const {username = '', handleLogout} = this.props;
+    const {anchorEl} = this.state;
+    const open = Boolean(anchorEl);
+    return (
+      <React.Fragment>
+        <IconButton aria-owns={username ? 'sidebar-menu' : null} aria-haspopup="true" color="inherit" onClick={this.handleMenu}>
+          <AccountCircle />
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: Also, display logged in username.

File: src/webui/components/Header/index.js [hunk 5]
```diff
@@ -1,69 +1,179 @@
...
+        ) : (
+          <Button color="inherit" onClick={this.handleToggleLogin}>
+            Login
+          </Button>
+        )}
+      </div>
+    );
+  }
+
+  renderMenu() {
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: renderLoggedInUserMenu

File: src/webui/components/Header/infoDialog/index.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import {IProps} from './interfaces';
+
+const InfoDialog = ({open = false, children, onClose}: IProps): ReactElement => (
+  <Dialog open={open} onClose={onClose}>
+    <Title disableTypography>Register Info</Title>
+    <Content>{children}</Content>
+    <DialogActions>
+      <Button onClick={onClose} color="inherit" autoFocus>
+        OK
```
Comment (ayusharma) [inline src/webui/components/Header/infoDialog/index.js]: It should be `Close`

File: src/webui/components/Header/interfaces.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @prettier
+ */
+
+// @flow
+
+export interface IProps {
+  username?: string;
+  handleLogout: () => {};
```
Comment (ayusharma) [inline src/webui/components/Header/interfaces.js]: Use `() => void`
https://flow.org/en/docs/types/functions/#toc-function-types

File: src/webui/components/Header/index.js [hunk 8]
```diff
@@ -1,69 +1,179 @@
...
+
+  renderLeftSide() {
+    return (
+      <Link to="/">
+        <Logo />
+      </Link>
+    );
+  }
+
+  renderRightSide() {
```
Comment (ayusharma) [inline src/webui/components/Header/index.js]: Here we can do some separation. Keep login/logput flow separate. Icons can be grouped in other function.

File: src/webui/components/Header/interfaces.js (hunk 7 above)
Comment (ayusharma) [reply src/webui/components/Header/interfaces.js]: or `handleLogout: Function`

File: src/webui/components/Header/index.js (hunk 3 above)
Comment (juanpicado) [reply src/webui/components/Header/index.js]: It will host more than that, `renderInfoDialog` is ok in this case.

**Comment by ayusharma** [comment]
> Type: Refactor
> 
> The following has been addressed in the PR:
> 
> [#955](https://github.com/verdaccio/verdaccio/issues/955)
> [#986](https://github.com/verdaccio/verdaccio/issues/986)
> 
> Description: "Create new App Bar "
> 
> * Done:
> 
> * [x]  Created css variables in JS.
> * [x]  Added JS helpers.
> * [x]  Replaced old sidebar by the "APP Bar" component from Material UI.
> * [x]  Replaced logo by the new version.
> * [x]  Added Avatar icon.
> * [x]  Added Info Icon.
> * [x]  Added Information Dialog, where the user can 'copy to the clipboard' the verdaccio commands.
> * [x]  Added Verdaccio website documentation link.
> * [x]  Added Drop Down Menu on the right side with a logout option
> 
> * To do:
> 
> * [ ]  Refactor Tests using Material UI helper (createShallow).
> * [ ]  Add search bar in the sidebar. It will not be displayed in mobile version.
> * [ ]  Add option to add avatar image (upload).
> * [ ]  Add option to change the logo (upload).
> * [ ]  Make it really responsive.
> * [ ]  Move all css variables to JS.
> * Add i18n. (Future ticket)
> 
> * Test:
> 
> * [x]  Check login modal UI tests

@priscilawebdev Most of the things done. Please pick the remaining tasks from https://github.com/verdaccio/verdaccio/projects/7#card-13265438

**Pull request merged**